src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -1413,7 +1439,7 @@ private static InitialSessionState CreateRestrictedForRemoteServer()
             }
 
             // Add built-in variables.
```

Comment:
`LocalRunspace.ResetRunspaceState()` adds all variables defined in `BuiltInVariables` when resetting the Runspace, which bypass the `AddVariables` method. However, we can probably ignore this for this PR, because I think that behavior of `ResetRunspaceState` is not correct. I opened #14987 for it.